### DISCUSSION TITLE
Centering lyrics

### DISCUSSION THREAD

**Discussion by NoahSylwester** [rism-digital/verovio#3067] [Q&A]
Hi, I'm working on adjusting some behavior in verovio to fit the needs of my own project. One thing that I'm trying to achieve is centering lyric parts on their corresponding notes. I've been mucking around with Syl::PrepareLyrics, and LayerElement::m_xAbs (maybe as a workaround),  LayerElement::m_drawingXRel to no real avail. Not necessarily looking for solutions here, but I'm hoping perhaps somehow might know enough to point me in the right direction. Also maybe an idiot-check if this feature already somehow exists somewhere without my knowledge.

Thanks in advance!

**Comment by lpugin** [comment]
Could you give a minimal MEI example encoding and the desired output?

**Comment by NoahSylwester** [comment]
Of course, thanks for any help. Here's a short song -- I've attached a zip of the musicxml file, which is the file format I am given for this.

The first screenshot is this file rendered by Finale, notice the lyrics are generally centered below the corresponding note. This is the behavior I'm aiming for. 

The second screenshot is how I'm able to render this file through verovio right now, notice how the lyrics are generally left-aligned with the corresponding note.

[French.musicxml.zip](https://github.com/rism-digital/verovio/files/9748102/French.musicxml.zip)
<img width="611" alt="Screen Shot 2022-10-10 at 11 14 45 AM" src="https://user-images.githubusercontent.com/18040774/194929184-f9096c29-32cb-486e-b511-5cffe889b078.png">
<img width="679" alt="Screen Shot 2022-10-10 at 11 17 59 AM" src="https://user-images.githubusercontent.com/18040774/194929375-20ec246d-cefb-4245-853f-9a2169d84d14.png">
